### PR TITLE
feat: Add closeAccount method

### PR DESCRIPTION
**Pull request by RamirezAlex**
Closes: #308

### CODE REVIEW

File: libs/api/account/data-access/src/lib/api-account-data-access.service.ts [hunk 1]
```diff
@@ -84,6 +107,60 @@ export class ApiAccountDataAccessService implements OnModuleInit {
     return solana.getTokenAccounts(accountId, mint.toString())
   }
 
+  async closeAccount(input: CloseAccountRequest): Promise<AppTransaction> {
```
Comment (RamirezAlex) [inline libs/api/account/data-access/src/lib/api-account-data-access.service.ts:110]: CloseAccount and CreateAccount are very similar, I guess we should also include them in the refactor we are planing in #292

File: libs/sdk/src/lib/kinetic-sdk-internal.ts [hunk 2]
```diff
@@ -65,6 +67,36 @@ export class KineticSdkInternal {
...
+  async closeAccount({ owner, mint }: CloseAccountOptions): Promise<AppTransaction> {
+    if (!this.appConfig) {
+      throw new Error(`AppConfig not initialized`)
+    }
+    mint = mint || this.appConfig.mint.publicKey
+    const { lastValidBlockHeight, latestBlockhash, mintFeePayer, mintPublicKey } = await this.prepareTransaction({
+      mint,
+    })
+
+    const tx = await serializeCreateAccountTransaction({
```
Comment (beeman) [inline libs/sdk/src/lib/kinetic-sdk-internal.ts:79]: I think this should call `serializeCloseAccountTransaction`?

File: libs/solana/src/lib/helpers/serialize-close-account-transaction.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { Keypair } from '@kin-kinetic/keypair'
+import { TransactionType } from '@kin-tools/kin-memo'
+import { generateKinMemoInstruction } from '@kin-tools/kin-transaction'
+import { createCloseAccountInstruction, getAssociatedTokenAddress, TOKEN_PROGRAM_ID } from '@solana/spl-token'
+import { Transaction, TransactionInstruction } from '@solana/web3.js'
+import { PublicKeyString } from '../interfaces'
+import { getPublicKey } from './get-public-key'
+
+export async function serializeCreateAccountTransaction({
```
Comment (beeman) [inline libs/solana/src/lib/helpers/serialize-close-account-transaction.ts:9]: I think this should be called `serializeCloseAccountTransaction`?

File: libs/api/account/data-access/src/lib/api-account-data-access.service.ts [hunk 4]
```diff
@@ -84,6 +107,60 @@ export class ApiAccountDataAccessService implements OnModuleInit {
...
+    })
+    const mint = appEnv.mints.find(({ mint }) => mint.address === input.mint)
+    if (!mint) {
+      this.closeAccountErrorMintNotFoundCounter.add(1, { appKey })
+      throw new Error(`Can't find mint ${input.mint} in environment ${input.environment} for index ${input.index}`)
+    }
+    const signer = Keypair.fromSecretKey(mint.wallet?.secretKey)
+
+    const { feePayer, source, transaction } = parseAndSignTransaction({
+      tx: Buffer.from(input.tx, 'base64'),
```
Comment (beeman) [inline libs/api/account/data-access/src/lib/api-account-data-access.service.ts:128]: Here we need to use the `serializeTransaction` function

File: libs/api/account/feature/src/lib/api-account-feature.controller.ts [hunk 5]
```diff
@@ -27,6 +28,14 @@ export class ApiAccountFeatureController {
     return this.service.getAccountInfo(environment, index, accountId, commitment)
   }
 
+  @Post('close')
+  @ApiBody({ type: CloseAccountRequest })
+  @ApiOperation({ operationId: 'closeAccount' })
+  @ApiResponse({ type: AppTransaction })
+  closeAccount(@Body() body: CloseAccountRequest) {
+    return this.service.createAccount(body)
```
Comment (beeman) [inline libs/api/account/feature/src/lib/api-account-feature.controller.ts:36]: This should call into the `closeAccount` method

File: libs/solana/src/lib/helpers/generate-close-account-transaction.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    type: TransactionType.None,
+  })
+
+  // Get AssociatedTokenAccount
+  const associatedTokenAccount = await getAssociatedTokenAddress(mintKey, signer.publicKey)
+
+  // Create Transaction
+  const instructions: TransactionInstruction[] = [
+    appIndexMemoInstruction,
+    createCloseAccountInstruction(TOKEN_PROGRAM_ID, associatedTokenAccount, feePayerKey, []),
```
Comment (beeman) [inline libs/solana/src/lib/helpers/generate-close-account-transaction.ts:32]: Please check the properties of this `createCloseAccountInstruction` 

image


We only need to pass in `account`, `destination` and `authority`.

**Pull request closed without merging**